server/src/main/java/io/pravega/schemaregistry/service/SchemaRegistryService.java

```diff
@@ -0,0 +1,924 @@
+                                   log.info("Returning groups {}", groupsList);
+                                   Map<String, GroupProperties> collect = groupsList.getValue().stream().collect(
+                                           Collectors.toMap(AbstractMap.SimpleEntry::getKey, x -> x.getValue().get()));
+                                   return new MapWithToken<>(
+                                           collect, groupsList.getKey());
+                               });
+    }
+
+    /**
+     * Creates new group. Idempotent behaviour. If group already exists, it returns false.
+     *
+     * @param namespace       namespace for which the request is scoped to.
+     * @param group           Name of group.
+     * @param groupProperties Group properties.
+     * @return CompletableFuture which is completed when create group completes. True indicates this was
+     * new group, false indicates it was an existing group.
+     */
+    public CompletableFuture<Boolean> createGroup(String namespace, String group, GroupProperties groupProperties) {
+        Preconditions.checkArgument(group != null);
+        Preconditions.checkArgument(groupProperties != null);
+        Preconditions.checkArgument(validateRules(groupProperties.getSerializationFormat(), groupProperties.getCompatibility()));
+        log.info("create group called for {} with group properties {}", group, groupProperties);
+        return store.createGroup(namespace, group, groupProperties)
+                    .whenComplete((r, e) -> {
+                        if (e == null) {
+                            if (r) {
+                                log.info("Group {} created successfully.", group);
+                            } else {
+                                log.info("Group {} exists.", group);
+                            }
+                        } else {
+                            log.warn("create group {} request failed with error", group, e);
+                        }
+                    });
+    }
+
+    /**
+     * Gets group's properties.
+     * {@link GroupProperties#serializationFormat} which identifies the serialization format used to describe the schema.
+     * {@link GroupProperties#compatibility} sets the schema compatibility policy that needs to be enforced for evolving schemas.
+     * {@link GroupProperties#allowMultipleTypes} that specifies multiple schemas with distinct {@link SchemaInfo#type} can
+     * be registered.
+     * {@link GroupProperties#properties} properties for a group.
+     *
+     * @param namespace namespace for which the request is scoped to.
+     * @param group     Name of group.
+     * @return CompletableFuture which holds group properties upon completion.
+     */
+    public CompletableFuture<GroupProperties> getGroupProperties(String namespace, String group) {
+        Preconditions.checkArgument(group != null);
+        log.info("getGroupProperties called for group {}.", group);
```

Comment:
`trace` or `trace request` here and in other methods, to avoid flooding of logs. Context should include namespace as well.

Reply:
added namespace in the logs. 
but schema registry is not going to receive a lot of requests. so logging at info is fine for requests its receiving. all the processing for the request is logged at debug to avoid log flooding